index.js

```diff
@@ -58,12 +43,6 @@ Geocoder.prototype = inherit(mapboxgl.Control, {
   onAdd: function(map) {
     this.request = new XMLHttpRequest();
 
-    this.container = this.options.container ?
-      typeof this.options.container === 'string' ?
-      document.getElementById(this.options.container) :
-      this.options.container :
-      map.getContainer();
```

Comment:
The ability to attach the geocoder to a DOM node outside of the map container is important.